src/yunohost/certificate.py

```diff
@@ -0,0 +1,717 @@
+            if domain not in yunohost_domains_list:
+                raise MoulinetteError(errno.EINVAL, m18n.n('certmanager_domain_unknown', domain=domain))
+
+    certificates = {}
+
+    for domain in domain_list:
+        status = _get_status(domain)
+
+        if not full:
+            del status["subject"]
+            del status["CA_name"]
+            status["CA_type"] = status["CA_type"]["verbose"]
+            status["summary"] = status["summary"]["verbose"]
+
+        del status["domain"]
+        certificates[domain] = status
+
+    return { "certificates" : certificates }
+
+
+def certificate_install(auth, domain_list, force=False, no_checks=False, self_signed=False):
+   
+    """
+    Install a Let's Encrypt certificate for given domains (all by default)
+
+    Keyword argument:
+        domain_list  -- Domains on which to install certificates
+        force        -- Install even if current certificate is not self-signed
+        no-check     -- Disable some checks about the reachability of web server
+                       before attempting the install
+        self-signed  -- Instal self-signed certificates instead of Let's Encrypt
+    """
+
+    # Check if old letsencrypt_ynh is installed
+    _check_old_letsencrypt_app()
+    
+
+    if self_signed:
+        certificate_install_selfsigned(domain_list, force)
+    else:
+        certificate_install_letsencrypt(auth, domain_list, force, no_checks)
+
+
+def certificate_install_selfsigned(domain_list, force=False):
+    for domain in domain_list:
+
+        # Check we ain't trying to overwrite a good cert !
+        status = _get_status(domain)
+
+        if status and status["summary"]["code"] in ('good', 'great') and not force:
+            raise MoulinetteError(errno.EINVAL, m18n.n('certmanager_attempt_to_replace_valid_cert', domain=domain))
+
+
+        date_tag = datetime.now().strftime("%Y%m%d.%H%M%S")
+        new_cert_folder = "%s/%s-history/%s-selfsigned" % (CERT_FOLDER, domain, date_tag)
+
+        os.makedirs(new_cert_folder)
+
+        # Get serial
+        ssl_dir = '/usr/share/yunohost/yunohost-config/ssl/yunoCA'
+        with open(os.path.join(ssl_dir, 'serial'), 'r') as f:
+            serial = f.readline().rstrip()
+
+        shutil.copyfile(os.path.join(ssl_dir, "openssl.cnf"), os.path.join(new_cert_folder, "openssl.cnf"))
+
+        # FIXME : should refactor this to avoid so many os.system() calls...
+        # We should be able to do all this using OpenSSL.crypto and os/shutil
+        command_list = [
+            'sed -i "s/yunohost.org/%s/g" %s/openssl.cnf' % (domain, new_cert_folder),
+            'openssl req -new -config %s/openssl.cnf -days 3650 -out %s/certs/yunohost_csr.pem -keyout %s/certs/yunohost_key.pem -nodes -batch 2>/dev/null'
+            % (new_cert_folder, ssl_dir, ssl_dir),
+            'openssl ca -config %s/openssl.cnf -days 3650 -in %s/certs/yunohost_csr.pem -out %s/certs/yunohost_crt.pem -batch 2>/dev/null'
```

Comment:
Instead of hidding stderr we should probably store them somewhere and show them if the command failed (and log them somewhere anyway), otherwise this would make debugging really hard when something bad happen.